File: problems_list/0162.寻找峰值/0162-寻找峰值.cpp
class Solution {
public:
    int findPeakElement(vector<int>& nums) {
        int l=0, r=nums.size()-1;
        while(l<r){
            int m=l+((r-l)>>1);
            if(nums[m]<nums[m+1]){
                l=m+1;
            }else{
                r=m;
            }
        }
        return l;
    }
};

File: problems_list/0162.寻找峰值/0162-寻找峰值.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findPeakElement = function(nums) {
    let l=0, r=nums.length-1;
    while(l<r){
        let m=(l+r)>>1;
        if(nums[m]<nums[m+1]){
            l=m+1;
        }else{
            r=m;
        }
    }
    return l;
};